lib/hanami/router.rb

```diff
@@ -878,6 +882,12 @@ def prefixed_name(name)
       @name_prefix.relative_join(name, PREFIXED_NAME_SEPARATOR).to_sym
     end
 
+    # @since 2.3.0
+    # @api private
+    def underscored_name(name)
```

Comment:
It doesn't look like prefixed_name is used anywhere else, and it's defined as a private method + private API, so can we just push the underscoring into that method? It'll save some coercion back-and-forth between symbol and string, but more importantly, it keeps thing simple and doesn't overly abstract helper methods.

Reply:
Absolutely, it now reads:

```ruby
    def prefixed_name(name)
      @name_prefix
        .relative_join(name, PREFIXED_NAME_SEPARATOR)
        .gsub(NAME_UNDERSCORE_REGEXP, "_")
        .to_sym
    end
```

Hope, the multi-line notation with leading dots  is okay, opinions and tastes differ on this one.

Rubocop appears to be okay with it, but it barks at the constants being declared following `private`:

> Useless private access modifier for constant scope.

How do you deal with such situations?

* Move `private` or exclude the cop rule?
* Mop this up as part of a PR like this one or do a fresh PR for it?

Reply:
I really don't like that rubocop rule. If we haven't disabled it yet, we should.